soh/src/code/code_800EC960.c

```diff
@@ -1550,9 +1550,11 @@ void func_800ED458(s32 arg0) {
     u32 phi_v1_2;
     bool dpad = CVar_GetS32("gDpadOcarinaText", 0);
 
-    if (D_80130F3C != 0 && D_80131880 != 0) {
-        D_80131880--;
-        return;
+    if (D_80130F3C != 0 && sOcarinaDropInputTimer != 0) {
+        sOcarinaDropInputTimer--;
+        if (CVar_GetS32 ("gDpadNoDropOcarinaInput", 0) == 0) {
```

Comment:
Nitpick: Odd formatting here with spaces, also `== 0` is less clear than just `!` imo